src/test/bench_test/benchmark.cpp

```diff
@@ -191,6 +235,58 @@ void benchmark::read_random(thread_arg *thread)
     thread->stats.add_message(msg);
 }
 
+void benchmark::multi_get_random(thread_arg *thread)
+{
+    uint64_t bytes = 0;
+    uint64_t found = 0;
+    int max_fetch_count = 100;
+    int max_fetch_size = 1000000;
+
+    for (int i = 0; i < FLAGS_benchmark_num / 100; i++) {
+
+        // generate hash key
+        std::string hashkey, sort_key, value;
+        hashkey = generate_string(FLAGS_hashkey_size);
+        std::map<std::string, std::string> kvs;
+        std::set<std::string> sortkeys;
+
+        // generate sort key
+        // generate value for random to keep in peace with write
+        for (int j = 0; j < 100; j++) {
+            sortkeys.insert(generate_string(FLAGS_sortkey_size));
+            value = generate_string(FLAGS_value_size);
+        }
+
+        // read from pegasus
+        int try_count = 0;
+        while (true) {
+            try_count++;
+            int ret = _client->multi_get(
+                hashkey, sortkeys, kvs, max_fetch_count, max_fetch_size, FLAGS_pegasus_timeout_ms);
+            if (ret == ::pegasus::PERR_OK) {
+                found += 100;
+                bytes += (FLAGS_hashkey_size + FLAGS_sortkey_size + FLAGS_value_size) * 100;
```

Comment:
The returned count and bytes depends on `kvs`, not the FLAGS_*, suppose the case when the database is empty or filled by different flags.